src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -4342,6 +4351,102 @@ private void copyAdditionalJavadocResources( File anOutputDirectory )
         return getPackageNamesOrFilesWithUnnamedPackages( sourcePaths, files, true );
     }
 
+    /**
+     * @param allSourcePaths     not null, containing absolute and relative paths
+     * @return a list of exported package names for files in allSourcePaths
+     * @throws MavenReportException if any
+     * @see #getFiles
+     * @see #getSourcePaths()
+     */
+    private List<String> getPackageNamesRespectingJavaModules( Map<String, Collection<String>> allSourcePaths )
+            throws MavenReportException
+    {
+        List<String> returnList = new ArrayList<>();
+
+        if ( !StringUtils.isEmpty( sourcepath ) )
+        {
+            return returnList;
+        }
+        LocationManager locationManager = new LocationManager();

```

Comment:
I hope we can refactor the code, so the locationManager is used only once per goal-call.

Reply:
This method only gets invoked once per goal-call. Anyway, feel free to refactor this code to your liking. I don't mind :)

Reply:
locationManager is used at least twice now, which is pretty expensive. I'll refactor it